R-package/NAMESPACE

```diff
@@ -52,6 +52,7 @@ importFrom(graphics,barplot)
 importFrom(graphics,par)
 importFrom(jsonlite,fromJSON)
 importFrom(methods,is)
+importFrom(parallel,detectCores)
```

Comment:
please add `parallel` to the `Imports:` section in `DESCRIPTION`, immediately following `methods`.

https://github.com/microsoft/LightGBM/blob/d163c2c16629e3da41dc5d7b8dbb76b8fc6249db/R-package/DESCRIPTION#L57-L64

Reply:
`parallel` is part of base R. Base packages are not supposed to be mentioned there.

Reply:
I believe, based on https://github.com/microsoft/LightGBM/pull/4972#discussion_r838027789, that `{parallel}` should be removed from this PR, but in case this comes up in the future I'll explain here the evidence in favor of putting it in `Imports`.

"Writing R Extensions", in the ["Package Dependencies"](https://cran.r-project.org/doc/manuals/r-release/R-exts.html#Package-Dependencies) sub-section in the section about `DESCRIPTION`, explicitly states that package authors should include "all the standard packages that are used".

> The ‘Imports’ field lists packages whose namespaces are imported from (as specified in the NAMESPACE file) but which do not need to be attached... Ideally this field will include all the standard packages that are used

The only package which it says not to include is `{base}`.

> It makes no sense to declare a dependence on R without a version specification, nor on the package **base**: this is an R package and package **base** is always available

As shown in the link above, `{lightgbm}` includes other default packages like `{graphics}` and `{utils}` in `Imports:`. Those have been there since the first CRAN release in October 2020, and CRAN has never complained about them.

The CRAN mirror on GitHub shows many packages on CRAN which list `{parallel}` in their `Imports`: https://github.com/search?p=2&q=org%3Acran+filename%3ADESCRIPTION+%27parallel%2C%27&type=Code.

Including packages written by R-core members, like `{iotools}` ([link](https://github.com/cran/iotools/blob/714368d3f0785d5e3e6b16d174c80cc4cc57be70/DESCRIPTION#L7)).

`@`-ing other maintainers so they're aware of this for the purpose of future PR reviews: @jmoralez @StrikerRUS @guolinke @shiyu1994